### ISSUE TITLE
Update on hcp table for addurls

### ISSUE THREAD

**Issue by TobiasKadelka** [datalad/datalad-crawler#50]
Hey, I am creating the HCP-Table for addurls at the moment and since my script for that is taking a while I want to give you a short update. The program is slow, because it needs to send a request to the s3-storage, so I used the "datalad ls -Lr s3://hcp-openaccess/HCP_1200/" and basically then just parse the information into a csv (For testing reasons I did not separate the dataset-levels with // so far).
Then I ran the command
`datalad addurls  ~/hcp_123420_example.csv  '{original_url}' '{subject}/{filename}' `

Which leads to addurl(ok) and metadata(ok) for every file
```
addurl(ok): /home/homeGlobal/tkadelka/hcp/hcp/102311/HCP_1200/102311/release-notes/Diffusion_unproc.txt (file) [from datalad; to 102311/HCP_1200/102311/release-notes/Diffusion_unproc.txt]
metadata(ok): /home/homeGlobal/tkadelka/hcp/hcp/102311/HCP_1200/102311/release-notes/Diffusion_unproc.txt (file) 
```
and in the end:
```
action summary:
  add (notneeded: 1)
  addurl (ok: 68)
  metadata (ok: 68)
  save (notneeded: 1)
```

Here my script. Obviously, I would have to clean it before real use and delete the hardcoded numbers, but I want to inform you, that it works for my examples.
If you have thoughts and ideas about this, I am happy to here them :)

```python
import sys
import subprocess
import csv
def main():
        # s3-bucket prefix
        urlPrefix = "s3://hcp-openaccess/HCP_1200/"

        # names of the columns
        header = [["original_url","subject","filename", "version"]]

        # create/open the table and write the header
        this_table = csv.writer(open("hcp_123420_example.csv", 'w'), delimiter=',')
        this_table.writerows( header )

        # read the IDs from HCP
        #output = subprocess.getoutput("datalad ls -Lr s3://hcp-openaccess/HCP_1200/")
        output = subprocess.getoutput("datalad ls -Lr s3://hcp-openaccess/HCP_1200/123420/release-notes/")
        list_of_files = output.split("\n")


        # parsing the information
        for file in list_of_files:
                if not "HCP_1200/" in file:
                        continue
                subject_ID = file[ 9 : 15 ]
                parts = file.split(" ")
                file_name = parts[0]
                orig_url = "s3://hcp-openaccess/" + file_name
                version = ""
                for part in parts:
                        if "ver:" in part:
                                version = part[ 4 : ]
                if version == "":
                        continue
                this_table.writerows([[orig_url, subject_ID, file_name, version]])
 main()
```

**Comment by TobiasKadelka** [comment]
[hcp_123420_example.txt](https://github.com/datalad/datalad-crawler/files/3500175/hcp_123420_example.txt)

Here the example-table. It is a txt because github told my, csv is not ok to upload.

**Comment by TobiasKadelka** [comment]
Hey everybody,
here a little update on the HCP-table.
I modified the script for downloading the meta-Information, so know it is way faster (even if it still takes some time).
The table is still not complete, but at the moment I already have over 3.740.000 Lines (one per line) which contain the first 288 subjects.
I also checked again, that this table can be used, like we plan it:
 
`(hirni) tkadelka@brainbfast in /data/BnB_USER/Kadelka on git:master!;
mkdir ADDURLS;
cd ADDURLS;
datalad create --force ADDURLS_example;
cd ADDURLS_example;
git annex initremote datalad type=external externaltype=datalad encryption=none
datalad addurls  /data/BnB_USER/Kadelka/hcp_table.csv '{original_url}' '{subject}//{filename}';
[INFO   ] Creating a new annex repo at /data/BnB_USER/Kadelka/ADDURLS/ADDURLS_example 
create(ok): /data/BnB_USER/Kadelka/ADDURLS/ADDURLS_example (dataset)                                                                                                                                               
initremote datalad ok
(recording state in git...)
[INFO   ] Creating a new annex repo at /data/BnB_USER/Kadelka/ADDURLS/ADDURLS_example/111211 
create(ok): 111211 (dataset)`

So I am pretty confident now, that this way of creating a datalad dataset for hcp at least works :)
